design/deploy.md

```diff
@@ -0,0 +1,98 @@
+# Deploy
+
+A proposal to add support for hosting service deployment to the Polymer CLI.
+
+## Goals
+- Provide new users with a fast way to get a project hosted online.
+- Provide advanced users with an initial bootstrap for more advanced deployments.
+- Demonstrate the [PRPL](https://www.polymer-project.org/2.0/toolbox/server) pattern where HTTP/2 Server Push is available.
+
+## Non-goals
+- Do not try to support every possible deployment environment and configuration, or even a lot of them. Just enough to get users started.
+
+## Design
+
+- Add an `--environment` flag.
+  - Every command knows that it's being run in the context of this environment.
+  - Under the `default` environment, all commands operate as they do today.
+  - Under other environments:
+     - `build` now outputs the boilerplate needed to deploy the project to a given service, in addition to the traditional app build output.
+     - `serve` runs an appropriate HTTP server that emulates the configured hosting environment, instead of polyserve.
+
+- Add an `environments` section to `polymer.json`.
+  - **name**: The name of this environment.
+  - **type**: The type of hosting environment: `app-engine`, `gh-pages`, etc.
+  - Additional type-specific configuration (examples below).
+  - We could also treat environments as a special case of the existing build concept, and include them in the `build` section.
```

Comment:
Yeah I agree, that a deployment would probably have a source or distribution folder that is meant to be copied and/or served as root.  The builds may produce those or in a future world with smart servers deployment would just declare that the repo root is that root.  I'd propose a `root` property here, but I don't know if that's as generally applicable as I am naively thinking.